### ISSUE TITLE
make -j fails

### ISSUE THREAD

**Issue by Naima2002** [gsampler9/gSampler#6]
make -j fails despite fulfilling the software requirements:

CUDA: 12.3
Pytorch: 2.1.2
GCC: 9.4.0

**Comment by gpzlx1** [comment]
Could you give some cmake logs?

**Comment by Naima2002** [comment]
Sure, thanks!
[cmake_logs.tar.gz](https://github.com/gsampler9/gSampler/files/14089462/cmake_logs.tar.gz)

**Comment by gpzlx1** [comment]
I'm sorry, but I haven't encountered that particular issue. However, you may want to consider trying the following command:

```bash
cmake -DCMAKE_CUDA_COMPILER:PATH=/usr/local/cuda/bin/nvcc ...other cmake options...
```

This suggestion is mentioned in a related discussion on Stack Overflow: [https://stackoverflow.com/questions/55044846/no-cmake-cuda-compiler-could-be-found](https://stackoverflow.com/questions/55044846/no-cmake-cuda-compiler-could-be-found)

A similar error message in you log is:

```bash
Compiling the CUDA compiler identification source file "CMakeCUDACompilerId.cu" failed.
Compiler: CMAKE_CUDA_COMPILER-NOTFOUND 
```

Feel free to explore the provided link for more insights into resolving the issue.

**Comment by Naima2002** [comment]
oh my cmake works but when I run make -j I get the following:

~/gSampler/build$ make -j
[  2%] Built target gtest
[ 10%] Built target glog_internal
[ 12%] Built target gtest_main
[ 14%] Built target gmock
[ 16%] Built target glog
[ 17%] Built target glogtest
[ 19%] Built target gmock_main
[ 21%] Built target cleanup_with_absolute_prefix_unittest
[ 24%] Built target stacktrace_unittest
[ 26%] Built target logging_unittest
[ 28%] Built target cleanup_immediately_unittest
[ 31%] Built target demangle_unittest
[ 35%] Built target utilities_unittest
[ 35%] Built target symbolize_unittest
[ 40%] Built target signalhandler_unittest
[ 40%] Built target stl_logging_unittest
[ 42%] Built target cleanup_with_relative_prefix_unittest
[ 43%] Building CXX object CMakeFiles/gs.dir/src/batch_graph.cc.o
[ 44%] Building CXX object CMakeFiles/gs.dir/src/bcast.cc.o
[ 45%] Building CXX object CMakeFiles/gs.dir/src/cpu/sort_indices.cc.o
[ 47%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/batch_concat.cu.o
[ 48%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/batch_relabel.cu.o
[ 49%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/batch_split.cu.o
[ 50%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/batch_split2.cu.o
[ 51%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/batch_unique.cu.o
[ 52%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/encoding_decoding.cu.o
[ 54%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/list_sampling_probs.cu.o
[ 55%] Building CUDA object CMakeFiles/gs.dir/src/cuda/batch/slicing.cu.o
In file included from /home/anaima/.local/lib/python3.8/site-packages/torch/include/torch/csrc/api/include/torch/torch.h:3,
                 from /home/anaima/gSampler/src/./bcast.h:6,
                 from /home/anaima/gSampler/src/bcast.cc:1:
/home/anaima/.local/lib/python3.8/site-packages/torch/include/torch/csrc/api/include/torch/all.h:4:2: error: #error C++17 or later compatible compiler is required to use PyTorch.
    4 | #error C++17 or later compatible compiler is required to use PyTorch.
      |  ^~~~~
[ 56%] Building CUDA object CMakeFiles/gs.dir/src/cuda/compact.cu.o
[ 57%] Building CUDA object CMakeFiles/gs.dir/src/cuda/format_transform.cu.o
In file included from /home/anaima/.local/lib/python3.8/site-packages/torch/include/c10/util/string_view.h:4,

**Comment by gpzlx1** [comment]
Your PyTorch version is too recent, requiring C++17. 
The default C++ version specified in the CMake file is C++14. 
Modifying the CMake file to switch from 14 to 17 should be sufficient.

**Comment by Naima2002** [comment]
Changing the cmake file to c++17 would result in errors with the cmake command. I resolved the issue by downgrading to CUDA 12.1 and upgrading the cmake version. Thank you for your help!

**Comment by Liu-rj** [comment]
If there is no further problem, I'll close this issue.
